### ISSUE TITLE
check dmi table may cause misjudgement

### ISSUE THREAD

**Issue by chenxinlong** [ShellCode33/VM-Detection#2]
https://github.com/ShellCode33/VM-Detection/blob/4fd05cb2aa479b68529a3a7a7ea8956bd4b08c41/vmdetect/linux.go#L19-L32

DMI id may contains `xen` string, but I'm sure it is not a virtual machine : 
```
root@node1:~# cat /sys/class/dmi/id/board_serial
07V122XEN6001010
```

**Comment by ShellCode33** [comment]
Thanks for your feedback, it seems to be a tricky situation that would require some thinking.
Unfortunately I don't have time to maintain this. However I'd be more than happy to review and merge any PR you submit.

**Comment by chenxinlong** [comment]
I noticed that there's many conditions to check if is running in a virtual machine, can I say remove the `checkDMITable()` check condition   still makes `IsRunningInVirtualMachine()` reliable ? 

https://github.com/ShellCode33/VM-Detection/blob/4fd05cb2aa479b68529a3a7a7ea8956bd4b08c41/vmdetect/linux.go#L167-L214

**Comment by ShellCode33** [comment]
Yes you can remove it on your end if you want, it will still detect most of the usual stuff. I don't want to completely remove it here though. The "real" solution **might** be to perform a case-sensitive comparison with `Xen` instead.
